### DISCUSSION TITLE
Error: SASL_SIGNATURE_MISMATCH: The server did not return the correct signature (Supabase + Drizzle + AuthJS)

### DISCUSSION THREAD

**Discussion by jyuter** [supabase/supabase#21349] [Questions]
Hi Everyone,
I'm working on a project using Supabase for the DB, Drizzle for an ORM, and AuthJs for authentication. I'm able to run `supabase link` without a problem, I can push local db migrations to Supabase, but when I try logging in with Google, I get the following error:

```
Error: SASL_SIGNATURE_MISMATCH: The server did not return the correct signature

[auth][error] AdapterError: Read more at https://errors.authjs.dev#adaptererror
[auth][cause]: Error: SASL_SIGNATURE_MISMATCH: The server did not return the correct signature
    at SASLFinal (webpack-internal:///(rsc)/./node_modules/.pnpm/postgres@3.4.3/node_modules/postgres/src/connection.js:482:87)
    at Authentication (webpack-internal:///(rsc)/./node_modules/.pnpm/postgres@3.4.3/node_modules/postgres/src/connection.js:452:206)
    at handle (webpack-internal:///(rsc)/./node_modules/.pnpm/postgres@3.4.3/node_modules/postgres/src/connection.js:339:728)
    at Socket.data (webpack-internal:///(rsc)/./node_modules/.pnpm/postgres@3.4.3/node_modules/postgres/src/connection.js:225:17)
    at Socket.emit (node:events:518:28)
    at addChunk (node:internal/streams/readable:559:12)
    at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
    at Readable.push (node:internal/streams/readable:390:5)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
    at TCP.callbackTrampoline (node:internal/async_hooks:130:17)
[auth][details]: {}

```

If it helps, here's my `Auth.tsx` file:
```
import NextAuth from "next-auth";
import GitHub from "next-auth/providers/github";
import Google from "next-auth/providers/google";
import Facebook from "next-auth/providers/facebook";
import Twitter from "next-auth/providers/twitter";
import { DrizzleAdapter } from "@auth/drizzle-adapter";
import { db } from "@/database/db";


export const {
  handlers: { GET, POST },
  auth,
} = NextAuth({
  providers: [GitHub, Google, Facebook, Twitter],
  adapter: DrizzleAdapter(db),
});
```

And here's my `Db.tsx`:
```
import { drizzle } from "drizzle-orm/postgres-js";
import postgres from "postgres";
const connectionString = process.env.DATABASE_URL;
const client = postgres(connectionString, { prepare: false });
export const db = drizzle(client);
```

I've seen elsewhere that the solution to `SASL_SIGNATURE_MISMATCH:` is to change passwords, but that hasn't helped in my situation.

Many thanks in advance!

**Comment by GaryAustin1** [comment]
At the top of this discussion (search drizzle) https://github.com/orgs/supabase/discussions/17817 is an example of a connection string for drizzle. Also some generic docs.  https://supabase.com/docs/guides/database/connecting-to-postgres#connecting-with-drizzle

Not sure if this was working for you and the Supavisor change broke it, or you are just starting out.

**Reply by jyuter** [reply]
Thank you! As per the code I shared, it's essentially identical with the only difference being the URL is in an `env` variable rather than in the `.tsx` file. Everything I did was through following the docs.

**Reply by Ndidi** [reply]
Did you get a solution, @jyuter? I'm seeing the same error

**Comment by tashicorp** [comment]
I fixed it by removing the square brackets around the password in the database URI env var 🤦🏻‍♀️

**Reply by GeorgeAviateur** [reply]
As simple as that! I spent one hour trying to solve this hahahaha. Thanks mate!

**Reply by Darkspace-sg** [reply]
blimey!

**Reply by zhuangshiliu37-boop** [reply]
what?!!!!!!

**Comment by mikkokaar** [comment]
In case this helps anyone. Had an @ sign in the middle of my sufficiently long password, didn't notice it, drizzle kit didn't care, but the postgres npm package did care. Perhaps a different regex used for parsing the connection string (?)

If you try to connect to the pooler with a wrong password, you will get the rather misleading SASL_SIGNATURE_MISMATCH

Also, you need to pass the supabase cert (which can be downloaded from supabase dashboard > settings > database, below the enforce ssl setting) to postgres so you don't get an error about SELF_SIGNED_CERT_IN_CHAIN

`const ca = fs.readFileSync('prod-ca-2021.crt').toString();
const client = postgres(connectionString, {
  max: 1,
  ssl: {
    rejectUnauthorized: true,
    ca
  },
  prepare: false
});`
